### DISCUSSION TITLE
What do we think about utilizing the Navigate back/forward feature?

### DISCUSSION THREAD

**Discussion by adnauseum** [TS-Tobias/obsidian-no-tabs#1] [General]
I can't remember what Obsidian was like before everything opened up in a tab, but I know for a fact that I used my shortcuts to do the following:
- <kbd>cmd</kbd> + <kbd>Shift</kbd>+<kbd> <kbd>[</kbd> to navigate back
- <kbd>cmd</kbd> + <kbd>Shift</kbd>+<kbd> <kbd>[</kbd> to navigate forward

Example:
- <kbd>cmd</kbd> + <kbd>n</kbd> open a new note.
- <kbd>cmd</kbd> + <kbd>n</kbd> to open _another_ a new note.
- <kbd>cmd</kbd> + <kbd>Shift</kbd>+<kbd> <kbd>[</kbd> to navigate back to the first new note.

I have two questions:
1. Is anything like this possible presently?
2. If not, would people be interested in this? I can make a PR for that, but I don't know if that's something everyone wants.

**Comment by TS-Tobias** [comment]
Yes, I absolutely agree that the history should ideally be preserved.

The challenge when developing the plugin was to figure out a way to maintain the default behavior when it comes to creating a new note. In Obsidian, if you press `cmd` + `n`, a new file will be created and opened in a new tab - the cursor/focus is then placed on the filename input field.

In order to preserve this behavior, I decided not to mess with the note creation process and instead simply close the previous tab after the new tab becomes the active leaf (which means that the previous tab's history is gone).

If I created the file programmatically to open it in the current tab, I would need to set the filename of the note beforehand. I was unfortunately unable to find a way to move the cursor to the filename field at the top of the note to simulate the default behavior.

**TLDR**
1. You could do this by programmatically creating the file and opening it inside the current active leaf. This approach has two downsides though:
	1. You would have to manually focus the filename input field at the top of the note to change the filename. I didn't find a way to place the focus on this input field through the API.
	2. I would have to give the new note a temporary filename when creating the file programmatically.
2. I'm not sure if you are aware of this, but **with the latest release of Obsidian, the developers added a new hotkey called `Create new note in current tab` which basically makes this plugin no longer necessary**. You can simply set this command to `cmd` + `n` and remove the binding for the `Create new note` hotkey.

**Reply by adnauseum** [reply]
Ack.

Thank you for the reply and all the information!

I didn't know about the new hotkey. That enables navigation forward/backward, so my concern is irrelevant now. Thanks again!
